params/config.go

```diff
@@ -255,30 +399,102 @@ type TrustedCheckpoint struct {
 type ChainConfig struct {
 	ChainID *big.Int `json:"chainId"` // chainId identifies the current chain and is used for replay protection
 
+	// HF: Homestead
 	HomesteadBlock *big.Int `json:"homesteadBlock,omitempty"` // Homestead switch block (nil = no fork, 0 = already homestead)
+	// Note: EIPs 2 and 8 were also included in this fork, but have not been distinguished individually in the code.
+	//
+	// DELEGATECALL
+	// https://eips.ethereum.org/EIPS/eip-7
+	EIP7FBlock *big.Int `json:"eip7FBlock,omitempy"`
 
+	// HF: DAO
 	DAOForkBlock   *big.Int `json:"daoForkBlock,omitempty"`   // TheDAO hard-fork switch block (nil = no fork)
 	DAOForkSupport bool     `json:"daoForkSupport,omitempty"` // Whether the nodes supports or opposes the DAO hard-fork
 
+	// HF: Tangerine Whistle
 	// EIP150 implements the Gas price changes (https://github.com/ethereum/EIPs/issues/150)
 	EIP150Block *big.Int    `json:"eip150Block,omitempty"` // EIP150 HF block (nil = no fork)
 	EIP150Hash  common.Hash `json:"eip150Hash,omitempty"`  // EIP150 HF hash (needed for header only clients as only gas pricing changed)
 
-	EIP155Block      *big.Int `json:"eip155Block,omitempty"`      // EIP155 HF block
-	EIP158Block      *big.Int `json:"eip158Block,omitempty"`      // EIP158 HF block
-	DisposalBlock    *big.Int `json:"disposalBlock,omitempty"`    // Bomb disposal HF block
-	SocialBlock      *big.Int `json:"socialBlock,omitempty"`      // Ethereum Social Reward block
-	EthersocialBlock *big.Int `json:"ethersocialBlock,omitempty"` // Ethersocial Reward block
-
-	ByzantiumBlock      *big.Int `json:"byzantiumBlock,omitempty"`      // Byzantium switch block (nil = no fork, 0 = already on byzantium)
+	// HF: Spurious Dragon
+	EIP155Block *big.Int `json:"eip155Block,omitempty"` // EIP155 HF block
+	EIP158Block *big.Int `json:"eip158Block,omitempty"` // EIP158 HF block, includes implementations of 158/161, 160, and 170
+	//
+	// EXP cost increase
+	// https://github.com/ethereum/EIPs/blob/master/EIPS/eip-160.md
+	// NOTE: this json tag (a.) varies from it's 'siblings', which have 'F's in them, and (b.) without the 'F' will vary from ETH implementations if they choose to accept the PR
+	// FIXME?
+	EIP160FBlock *big.Int `json:"eip160Block,omitempty"`
```

Comment:
@sorpaas PTAL  This json tag business is a bit of a sticky situation.

The root cause of this is that the EIP158 namespace was used to symbolize a hardfork, including 3 separate EIPs (erm... none of which ended being named EIP158, to futher complicate the matter). 
So articulating params/features by an `EIPxxx` namespace gets messy. This was the whole reason I moved to using the `F` in the first place, where "F" stands for Feature.

Possible solves include...
- use without 'F' here, break json naming pattern, stay backwards
compatible for existing multi-geth clients, but incompatible
with ETH if they accept proposed changes as-is
- use with 'F' here, break backwards compatibility with
existing multi-geth clients, but be compatible with ETH as proposed
- push change without 'F' to ETH for review, keeping
in mind that it makes no sense for them to tangle their
code b/c of an establish multi-geth pattern
- push change without 'F' for any of the EIP160,161,170 to
eth. still breaks json naming pattern
- any other you can think of?...

Reply:
I have a slight preference that we use without `F`, *and* remove all grouped hard fork parameters (`ConstantinopleBlock`, `ByzantiumBlock`, etc).

* 1.8.20 will be the last version in the 1.8 series, meaning we can do breaking changes.
* Keeping both grouped features (`ConstantinopleBlock` etc) and individual features together (`EIP1234Block` etc) add maintenance burden, because we always need to check two booleans instead of one.
* For config parameters, keeping it in line with upstream may not be that important. The issue is that every time when config parameter changes, we need to resolve merge conflicts manually anyway.

Reply:
Roger that. 

Agreed on the `F` choice; I'll leave that as-is then. 

I'm looking at the implications of `IsConstantinople`/`ConstantinopleBlock` changes now. I definitely understand the motivations for removing them, and the appeal of a through-and-through consistent configuration pattern is enticing. My biggest hesitation is that using "groups" gives a way to "alias" often-reused configuration bunches, and removing them will require adding quite a few lines of repeated configurations which is both harder to read and, without adequate comments, possibly harder to understand....  I'm going to do a little exploring to see what the implications would look like.